docs/preparing_recipes_for_conan2.md

```diff
@@ -0,0 +1,368 @@
+from Conan 1.X to 2.0, please read the [migration
+guide](https://docs.conan.io/en/latest/conan_v2.html) in the Conan documentation.
+
+This document is a practical guide, offering extended information particular to Conan
+Center Index recipes to get them ready to upgrade to Conan 2.0.
+
+## New cpp_info set_property model
+
+New Conan generators like
+[CMakeDeps](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html)
+and
+[PkgConfigDeps](https://docs.conan.io/en/latest/reference/conanfile/tools/gnu/pkgconfigdeps.html),
+don't listen to *cpp_info* ``.names``, ``.filenames`` or ``.build_modules`` attributes.
+There is a new way of setting the *cpp_info* information with these
+generators using the ``set_property(property_name, value)`` method.
+
+All the information in the recipes, already set with the legacy model, should be
+translated to the new model. These two models **will live together in recipes** to make
+recipes compatible **with both new and legacy generators** for some time. After a stable
+Conan 2.0 version is released, and when the moment arrives that we don't support the
+legacy generators anymore in Conan Center Index, those attributes (``.names``,
+``.filenames`` etc.) will disappear from recipes, and only ``set_property`` methods will
+stay.
+
+We will cover some cases of porting all the information set with the legacy model to the
+new one. To read more about the properties available for each generator and how the
+properties model work, please check the [Conan documentation](https://docs.conan.io/en/latest/conan_v2.html#editables-don-t-use-external-templates-any-more-new-layout-model).
+
+> ⚠️ **Note**: Please, remember that the **new** ``set_property`` and the **legacy** attributes
+> model are *completely independent since Conan 1.43*. Setting ``set_property`` in recipes will
+> not affect CMake legacy generators (``cmake``, ``cmake_multi``, ``cmake_find_package`` and
+> ``cmake_find_package_multi``) at all.
+
+### CMakeDeps
+
+### Update required_conan_version to ">=1.43.0"
+
+If you set the property ``cmake_target_name`` in the recipe, the Conan minimum
+required version should be updated to 1.43. 
+
+```python
+
+required_conan_version = ">=1.43.0"
+
+class GdalConan(ConanFile):
+    name = "gdal"
+    ...
+```
+
+The reason for this change is that in Conan versions previous to 1.43 the
+``cmake_target_name`` values were not the final CMake target names. Those values were
+completed by Conan, adding namespaces automatically the final target names. After 1.43
+``cmake_target_name`` sets the **complete target name** that is added to the ``.cmake``
+files generated by Conan. Let's see an example:
+
+```python
+class GdalConan(ConanFile):
+    name = "gdal"
+    ...
+    def package_info(self):
+        # Before 1.43 -> Conan adds GDAL:: namespace -> Creates target with name GDAL::GDAL
+        self.cpp_info.set_property("cmake_target_name", "GDAL")
```

Comment:
IMHO, it is confusing to add this as _live_ code. We are not interested in <1.43, so I will add it as a comment, or don't add it at all.